templates/pkg/resource/sdk.go.tpl

```diff
@@ -237,15 +242,27 @@ func (rm *resourceManager) updateConditions (
 		}
 	}
 
+{{- if $reconcileRequeuOnSuccessSeconds := .CRD.ReconcileRequeuOnSuccessSeconds }}
+	if syncCondition == nil && onSuccess {
+		syncCondition = &ackv1alpha1.Condition{
+			Type:   ackv1alpha1.ConditionTypeResourceSynced,
+		}
+		syncCondition.Status = corev1.ConditionTrue
+		ko.Status.Conditions = append(ko.Status.Conditions, syncCondition)
+	}
+{{- else }}
+	// Required to avoid the "declared but not used" error in the default case
+	syncCondition = nil
```

Comment:
This will impact line 261, maybe a `_ = syncCondition` ?

Reply:
definitely cleaner this way even if line 261 wasn't an issue, fixed.